src/Sulu/Bundle/TestBundle/Testing/JsonRequestTrait.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\TestBundle\Testing;
+
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Component\DomCrawler\Crawler;
+
+trait JsonRequestTrait
+{
+    /**
+     * @param mixed[] $parameters
+     * @param mixed[] $server
+     */
+    protected static function jsonRequest(KernelBrowser $client, string $method, string $uri, array $parameters = [], array $server = [], bool $changeHistory = true): Crawler
+    {
+        $content = null;
+
+        if ('GET' !== $method) {
+            // JSON_THROW_ON_ERROR requires at least php 7.3
+            $content = \json_encode($parameters, \defined('JSON_THROW_ON_ERROR') ? \JSON_THROW_ON_ERROR : 0);
+            $parameters = [];
```

Comment:
I don't really get what's happening here... Is it logical to the user calling this function? Wouldn't it maybe be cleaner to separate `$content` and `$parameters`?

Reply:
`GET` can not have content so and are added as query parameter in symfony when its a GET request. When its POST or request Methods its converted by symfony into request parameter. Here is the difference now that we convert it into the content instead of request parameter. I will add a comment here so its clear.

Reply:
But what if we want to do a `POST` request with query parameters and some content in the body? It's probably still possible by adding it as a string to the url like `/admin?test=value`, but the `$parameters` parameter would not work for that, even though I would have assumed that, since Symfony in general is working that way. For these reasons I would split the `$parameters` and `$content`, so that it works the same as in Symfony (you have also mentioned that the parameters are the same as in the Symfony request, so they should also work in the same way IMO.

Reply:
@danrot also before when using POST query parameters has to be added as `?test=123` as symfony converts `$parameters` into request parameters. Did update to use the same logic as symfony and linked the code. So it should be more clear now what happens here that instead of converting to request it is converted into json for the specific methods.